rust/abacus-core/src/types/chain_data.rs

```diff
@@ -0,0 +1,57 @@
+use ethers::prelude::{H256, U256};
+
+/// Info about a given block in the chain.
+#[derive(Debug)]
+pub struct BlockInfo {
+    /// Hash of this block
+    pub hash: H256,
+    /// Unix timestamp of this block's creation in seconds
+    pub timestamp: u64,
+    /// Block height or the nth block in the chain
+    pub number: u64,
+    /// Total used gas by transactions in this block
+    pub gas_used: U256,
+    /// Maximum amount of gas allowed in this block
+    pub gas_limit: U256,
+}
+
+/// Information about a given transaction in the chain.
+#[derive(Debug)]
+pub struct TxnInfo {
```

Comment:
Ditto? Seems like for the explorer we only need the hash, sender, and maybe the recipient

Reply:
See previous

Reply:
These ones I see as potentially being _slightly_ more useful, mostly because we're on chains that Dune may not support. I'm a bit worried that by including so much ethereum-specific info now it could significantly slow us down when we support non-ethereum-based chains in the future.

How easy do you think it would be to do either of these?
1. If kept as it is now, change things to support different execution environments properly
2. To cut down to only basic properties here, and if in the future we decide we want this, backfill the data

Reply:
I think (1) is better, worst case we can shove in nulls to the database. I anticipate we will need enums to distinguish between different blockchain types in the future anyway and this would be trivially solved using one.

Reply:
Ok, given it's already been done I'm happy to keep as is. In the worst case if we have issues supporting a different execution environment in the future and we don't have a use for this type of data, we can always just scrap it later